qiskit/quantum_info/operators/symplectic/clifford.py

```diff
@@ -147,56 +173,102 @@ def __str__(self):
 
     def __eq__(self, other):
         """Check if two Clifford tables are equal"""
-        return super().__eq__(other) and self._table == other._table
+        return super().__eq__(other) and self._paulis == other._paulis
 
     # ---------------------------------------------------------------------
     # Attributes
     # ---------------------------------------------------------------------
     def __getitem__(self, key):
         """Return a stabilizer Pauli row"""
-        return self._table.__getitem__(key)
+        return self._paulis.__getitem__(key)
 
     def __setitem__(self, key, value):
         """Set a stabilizer Pauli row"""
-        self._table.__setitem__(key, value)
+        self._paulis.__setitem__(key, value)
 
     @property
+    def paulis(self):
+        """Return PauliList"""
+        return self._paulis
+
+    @paulis.setter
+    def paulis(self, value):
+        """Set the Pauli list"""
+        if not isinstance(value, PauliList):
+            value = PauliList(value)
+        self._paulis._z[:, :] = value._paulis._z
+        self._paulis._x[:, :] = value._paulis._x
+        self._paulis._phase[:] = value._paulis._phase
+
+    # TODO: DEPRECATE table, stabilizer, and destabilizer
+    @property
+    # @deprecate_function(
+    #    "The Clifford.table method is deprecated as of Qiskit Terra 0.19.0 "
+    #    "and will be removed no sooner than 3 months after the releasedate. "
+    #    "Use Clifford.paulis method instead.",
+    # )
     def table(self):
         """Return StabilizerTable"""
-        return self._table
+        return StabilizerTable(
+            np.column_stack((self.paulis.x, self.paulis.z)), phase=self.paulis.phase // 2
+        )
 
     @table.setter
+    # @deprecate_function(
+    #    "The Clifford.table method is deprecated as of Qiskit Terra 0.19.0 "
+    #    "and will be removed no sooner than 3 months after the releasedate. "
+    #    "Use Clifford.paulis method instead.",
+    # )
     def table(self, value):
         """Set the stabilizer table"""
         # Note this setter cannot change the size of the Clifford
         # It can only replace the contents of the StabilizerTable with
         # another StabilizerTable of the same size.
         if not isinstance(value, StabilizerTable):
             value = StabilizerTable(value)
-        self._table._array[:, :] = value._table._array
-        self._table._phase[:] = value._table._phase
+        self.paulis(value)
 
     @property
+    # @deprecate_function(
+    #    "The Clifford.stabilizer method is deprecated as of Qiskit Terra 0.19.0 "
+    #    "and will be removed no sooner than 3 months after the releasedate. "
+    #    "Use Clifford.paulis method instead.",
+    # )
     def stabilizer(self):
```

Comment:
I think that a `stabilizer` method is still useful since otherwise we should call: 
`clifford.paulis[num_qubits : 2 * num_qubits]` which is more complicated.
(especially in the `StabilizerState` and decomposition)

Reply:
We can't change the output type of the function without the deprecation period that's only starting now, anyway.